### PR TITLE
Generic Refactoring to utilise f-strings

### LINKED ISSUE #303: Generic Refactoring
### Current Behaviour:
Some code is not readable and some optimization should be applied. For example:
* URL formatting should be using: `str.format()`: `'{}/{}/{}'.format(..., ..., ...)` or better `f'{domain}/{path}/{id}'`
* issue #301
* please @chrizandr @xadahiya add to the list
* Adding doc strings wherever missing

### Expected Behaviour:
Code should be refactored

### PR DESCRIPTION
**Pull request by de-sh**
Fixes #303

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.6x and above.

### Description

Converted all string additions in repo into f-strings as planned. Works(Used Pipenv).

### Test Logs

```
Creating a temporary database...
going for create doc
Classes and properties added successfully.
Setting up Hydrus utilities... 
Creating utilities context... 
Setup done, running tests...
..../demoapi/anotherSingleClass
{'@context': '/demoapi/contexts/anotherSingleClass.jsonld', '@id': '/demoapi/anotherSingleClass', '@type': 'anotherSingleClass', 'Prop1': 'EGM6OU'}
/demoapi/singleClass
{'@context': '/demoapi/contexts/singleClass.jsonld', '@id': '/demoapi/singleClass', '@type': 'singleClass', 'Prop1': 'MAPQIF', 'Prop2': 'L1N1WZ', 'dummyProp': '/demoapi/DcTest/5fe633af-d1f0-411c-b7f1-622ecfcb47b3', 'singleClassProp': '/demoapi/anotherSingleClass/'}
.............Creating a temporary datatbase...
Classes, Properties and Users added successfully.
Setting up Hydrus utilities... 
Creating utilities context... 
Setup done, running tests...
..........Creating a temporary datatbsse...
['User', 'Pet']
Pet
Classes and properties added successfully.
Setup done, running tests...
.........SELECT EXISTS (SELECT * 
FROM instances 
WHERE instances.id = ?) AS anon_1
.......Importing Open Api Documentation ..
......Method on path /pet/findByStatus already present !
Method on path /pet/findByTags already present !
Method on path /pet/{petId} already present !
Cannot parse path /pet/{petId} because no ref to local class provided
Cannot parse path /pet/{petId} because no ref to local class provided
Cannot parse path /store/inventory because no ref to local class provided
Cannot parse path /store/order/{orderId} because no ref to local class provided
Method on path /user/createWithArray already present !
Method on path /user/createWithList already present !
Cannot parse path /user/login because no ref to local class provided
Cannot parse path /user/logout because no ref to local class provided
Cannot parse path /user/{username} because no ref to local class provided
...
----------------------------------------------------------------------
Ran 52 tests in 1.384s

OK
```

### Changes log
* Refactored regular string addition expressions into f-strings.

#### Added

NONE

#### Changed

- Change 1: Added f-strings in place of string addition

#### Fixed

- Bug 1: Personally created NameError

#### Removed

NONE

P.S: Please help me complete this PR as I am a bit new to this. Will make edits with due advice.

### CODE REVIEW

File: hydrus/resources.py [hunk 1]
```diff
@@ -494,14 +490,14 @@ def get(self, category: str) -> Response:
...
             return set_response_headers(jsonify(response))
         # Check for non collection class
         elif category in get_doc().parsed_classes:
-            response = {
+            response={
                 "@context": get_doc().parsed_classes[category]["context"].generate()}
             return set_response_headers(jsonify(response))
         else:
-            response = {404: "NOT FOUND"}
+            response={404: "NOT FOUND"}
```
Comment (vddesai1871) [inline hydrus/resources.py]: Do not remove spaces surrounding operators, binary operators should have a single space on either side to facilitate readability.

Comment (de-sh) [reply hydrus/resources.py]: Oh, ok. I ran a pep8 checker which was the reason for the change.

File: hydrus/tests/test_app.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def gen_dummy_object(class_, doc):
                 object_[prop.title] = gen_dummy_object(prop_class, doc)
             else:
                 object_[prop.title] = ''.join(random.choice(
-                    string.ascii_uppercase + string.digits) for _ in range(6))
+                    string.ascii_uppercase  string.digits) for _ in range(6))
```
[the remark this answers is not in the document]
Comment (de-sh) [reply hydrus/tests/test_app.py]: Would be wonderful if you could guide me through running a test. As I have pointed out, I am new to the concept of testing and not very accustomed to the practice. Sorry if this seems like a noob problem.

[the remark this answers is not in the document]
Comment (vddesai1871) [reply hydrus/tests/test_app.py]: Use `python -m unittest discover` to run all the tests

**Comment by de-sh** [comment]
The errors are only prevalent in python<3.6 due to them lacking the config for f-strings. I might have to retract on this whole PR in that case.

**Comment by Mec-iS** [comment]
If I remember well, string formatting with `f''` is supported only from Python 3.6 onward. We want the codebase to be compatible with Python 3.5.2 onward.

**Comment by de-sh** [comment]
@Mec-iS, please checkout #314. I'll close this one, is that okay?

**Pull request closed without merging**